### PR TITLE
Re-add file sort by "last date read"

### PR DESCRIPTION
**Pull request by snelg**
Partial un-do of #10627 : `access` is not the same as `modified`, so we want to keep the "last read date" file sort option in addition to the "date modified" option

- - -
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader/10682)

### CODE REVIEW

**Comment by NiLuJe** [comment]
See https://github.com/koreader/koreader/pull/10627#issuecomment-1631708476

----

I *can* confirm that behavior, but I'm slightly confused.

Do *we* touch the file or timestamps ourselves somewhere?

When I tested the atime things the first time we decided to kill it, I was doing synthetic tests w/ stuff like touch, less & vim. And that didn't do much, as expected (noatime & co).

But, if I actually *open* a book in KOReader (doesn't matter if CRe or PDF), we do affect both its atime and ctime...

```
┌─(niluje@ahsoka:pts/16)───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────(/var/tmp/niluje/koreader/koreader-emulator-x86_64-pc-linux-gnu-debug/koreader)─┐
└─(0.17:18%:03:54:%)── stat /var/tmp/niluje/koreader/test/juliet.epub                                                                                                                                                                                                                                                                                  ──(Wed, Jul 12)─┘
  File: /var/tmp/niluje/koreader/test/juliet.epub
  Size: 839144          Blocks: 1640       IO Block: 4096   regular file
Device: 0,33    Inode: 1262789     Links: 1
Access: (0644/-rw-r--r--)  Uid: ( 1000/  niluje)   Gid: ( 1000/  niluje)
Access: 2023-07-12 03:12:38.000000000 +0200
Modify: 2023-07-12 02:44:23.000000000 +0200
Change: 2023-07-12 03:12:38.533349776 +0200
 Birth: 2023-07-12 02:44:23.825471810 +0200
┌─(niluje@ahsoka:pts/16)───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────(/var/tmp/niluje/koreader/koreader-emulator-x86_64-pc-linux-gnu-debug/koreader)─┐
└─(0.17:18%:03:54:%)── stat /var/tmp/niluje/koreader/test/juliet.epub                                                                                                                                                                                                                                                                                  ──(Wed, Jul 12)─┘
  File: /var/tmp/niluje/koreader/test/juliet.epub
  Size: 839144          Blocks: 1640       IO Block: 4096   regular file
Device: 0,33    Inode: 1262789     Links: 1
Access: (0644/-rw-r--r--)  Uid: ( 1000/  niluje)   Gid: ( 1000/  niluje)
Access: 2023-07-12 03:54:42.000000000 +0200
Modify: 2023-07-12 02:44:23.000000000 +0200
Change: 2023-07-12 03:54:42.496024101 +0200
 Birth: 2023-07-12 02:44:23.825471810 +0200
```

**Comment by NiLuJe** [comment]
> Do _we_ touch the file or timestamps ourselves somewhere?

Okay, mystery solved ;).

https://github.com/koreader/koreader/blob/d57325aaf6c102a0f57e0706b60b96579827728c/frontend/readhistory.lua#L263

TL;DR: Yeah, this makes sense, and both that and this PR are desirable.

**Review by NiLuJe — APPROVED** [review]

File: frontend/ui/widget/filechooser.lua [hunk 1]
```diff
@@ -340,7 +344,9 @@ function FileChooser:getMenuItemMandatory(item, collate)
     local text
     if collate then -- file
         -- display the sorting parameter in mandatory
-        if collate == "date" then
+        if collate == "access" then
+            text = os.date("%Y-%m-%d %H:%M", item.attr.access)
```
Comment (hius07) [inline frontend/ui/widget/filechooser.lua:348]: Thank you. Can you use the `datetime` function to format the date.

Comment (snelg) [reply frontend/ui/widget/filechooser.lua]: Sure. It bugs me just a very, very tiny amount that the formatted date has two spaces if the hour value is under 10, e.g.
`2023-07-11  4:00 PM`
instead of
`2023-07-11 4:00 PM`

But I don't know what else that might break if that changed. A quick scan looks like those `secondsTo[X]` functions are all for display purposes only, but I'm not 100% sure they're not used for comparison somewhere I didn't notice.

**Comment by snelg** [comment]
And to be clear, I don't mind if the proper solution is to make it so the `mtime` updates each time a file is opened, thus making the "date modified" the only needed Date-sorting option.
My patch here was just the easiest code change I saw since it simply un-does a few line changes.

File: frontend/ui/widget/filechooser.lua (hunk 1 above)
Comment (NiLuJe) [reply frontend/ui/widget/filechooser.lua]: That's very much on purpose for display/alignment reasons, IIRC, yeah ;).

**Comment by hius07** [comment]
I agree, it makes sense to have both of them.

**Review by hius07 — APPROVED** [review]

**Review by Frenzie — APPROVED** [review]

**Review by poire-z — APPROVED** [review]

**Comment by NiLuJe** [comment]
(Rebased it to make GH happy)
